src/typed-method-types/workflows/triggers/mod.ts

```diff
@@ -30,44 +58,90 @@ export type BaseTrigger = {
   description?: string;
 };
 
+// A helper to make sure inputs are passed. Required for automated triggers
+export type RequiredInputs = Required<
+  Pick<BaseTrigger, "inputs">
+>;
+
+/** @description Response object content for delete method */
+type DeleteResponse = {
+  ok: true;
+};
+
+type DeleteTriggerResponse = Promise<
+  DeleteResponse | FailedTriggerResponse
+>;
+
+type ListArgs = {
+  /** @description Lists triggers only if they owned by the caller */
+  is_owner?: boolean;
+  /** @description Lists triggers only if they have been published */
+  is_published?: boolean;
+};
+
+type ValidTriggerObjects =
+  | ShortcutTriggerObject<ShortcutTrigger>
+  | EventTriggerObject<EventTrigger>
+  | ScheduledTriggerObject<ScheduledTrigger>
+  | WebhookTriggerObject<WebhookTrigger>;
+
+type ListResponse = {
+  ok: true;
+  triggers: ValidTriggerObjects[];
+};
+
+type ListTriggerResponse = Promise<
+  ListResponse | FailedListResponse
+>;
+
+export type FailedListResponse = BaseResponse & {
+  ok: false;
+  /** @description no triggers are returned on a failed response */
+  triggers?: never;
+};
+
+export type FailedTriggerResponse = BaseResponse & {
+  ok: false;
+  /** @description no trigger is returned on a failed response */
+  trigger?: never;
+};
+
 export type TriggerIdType = {
   /** @description The id of a specified trigger */
-  trigger_id: string;
+  trigger_id: string | undefined;
 };
 
-// A helper to make sure inputs are passed. Required for automated triggers
-export type RequiredInputs = Required<Pick<BaseTrigger, "inputs">>;
-
 export type ValidTriggerTypes =
   | EventTrigger
   | ScheduledTrigger
   | ShortcutTrigger
   | WebhookTrigger;
 
-type BaseTriggerResponse = Promise<BaseResponse>;
-type TriggerResponse = BaseTriggerResponse;
-type ListArgs = {
-  /** @description Lists triggers only if they owned by the caller */
-  is_owner?: boolean;
-  /** @description Lists triggers only if they have been published */
-  is_published?: boolean;
+/** @description Overload function type for create method */
+type createType = {
+  <TriggerDefinition extends ValidTriggerTypes>(
+    args: BaseMethodArgs & TriggerDefinition,
+  ): ResponseTypes<TriggerDefinition>;
+};
+
+/** @description Overload function type for update method */
+type updateType = {
+  <TriggerDefinition extends ValidTriggerTypes>(
+    args: BaseMethodArgs & TriggerDefinition & TriggerIdType,
+  ): ResponseTypes<TriggerDefinition>;
```

Comment:
nit: The camel casing here feels inconsistent, should we use pascal case instead?
nit: I don't think these are overloads, they just seem like the typing for these methods has been extracted into a named type